k-distribution/include/kframework/builtin/domains.md

```diff
@@ -159,8 +159,8 @@ of any of the syntax defined in any of these modules.
 
   rule makeArray(I::Int, D::KItem) => arr(.List, I, D)
 
-  rule arr(L::List, _, _       ) [ IDX::Int ] => L[IDX] requires IDX <Int size(L)
-  rule arr(_      , _, D::KItem) [ _        ] => D      [owise]
+  rule arr(L::List, _, _       ) [ IDX::Int ] => L[IDX] requires 0 <=Int IDX andBool IDX  <Int size(L)
+  rule arr(L::List, _, D::KItem) [ IDX::Int ] => D      requires 0 <=Int IDX andBool IDX >=Int size(L)
```

Comment:
These rules don't cover the case that `IDX <Int 0` though, which we should cover I think to keep it total. I would stick with the `owise` rule.